### ISSUE TITLE
Extending placeholders which have nested media queries leaks placeholder into CSS output

### ISSUE THREAD

**Issue by keithamus** [sass/libsass#316]
### Abstract

Using a placeholder selector, calling a mixin which wraps a media-query, the placeholder selector is leaked into the CSS output.

### Detail

Given the following SCSS:

```scss
%foo {
	@media screen and (min-width: 300px) {
		max-width: 80%;
	}
}

body {
	@extend %foo;
}
```

#### Expected

The output of sass (3.3.1 and 3.2.15)

```css
@media screen and (min-width: 300px) {
  body {
    max-width: 80%; } }
```

#### Actual

The output of libsass (actually, node-sass 0.8.3 which is based off of hcatlin/libsass@1122ead208a8d1c438daaca70041ef6dd2361fa0)

```css
@media screen and (min-width: 300px) {
  %foo {
    max-width: 80%; } }
```

As you can see the placeholder `%foo` should be `body` but it is not. This is also the same case with multiple selectors extending off of %foo, it still displays %foo instead of the multiple classes that were extended.

I'm quite sure this is only media-query based, I have a reduced test-case with mixins:

```scss
@mixin test() {
	@content;
}
%foo {
	@include test() {
		max-width: 80%;
	}
}
body {
	@extend %foo;
}
```

The output of this is identical in SASS (3.3.1 and 3.2.15) and hcatlin/libsass@1122ead208a8d1c438daaca70041ef6dd2361fa0:

```
body {
  max-width: 80%; }
```

Further more, multiple mixins also work the same - for example:

```sass
@mixin test2() {
	@content
}
@mixin test() {
	@include test2() {
		max-width: 80%;
	}
}
%foo {
	@include test();
}
body {
	@extend %foo;
}
```

Produces the same output.

**Comment by mgreter** [comment]
This seems to work with latest master! //CC @xzyfer

**Comment by xzyfer** [comment]
Kind of not really @mgreter . The latest master produce the _expected_ output

```css
@media screen and (min-width: 300px) {
  body {
    max-width: 80%; } }
```

But not the correct output which is an error message.

**Comment by xzyfer** [comment]
Never mind my previous comment I was confusing this issue with https://github.com/sass/libsass/issues/317

**Comment by xzyfer** [comment]
The spec for this is in https://github.com/sass/sass-spec/tree/master/spec/regressions. Do you know what the folder is about @mgreter ?

**Comment by mgreter** [comment]
Not really. IMHO I recovered this via `perl-libsass` when `sassc` did not yet have the ability to discover passing tests itself. So I just copied that folder from todo!

**Comment by xzyfer** [comment]
I returned the _regression_ specs to their appropriate folders https://github.com/sass/sass-spec/pull/246.

**Comment by xzyfer** [comment]
This is fixed and will be 3.2.
